src/modules/clustering/hierarchical_clustering_method.cpp

```diff
@@ -117,6 +135,11 @@ std::vector<Cluster> hierarchical_clustering_method(std::vector<Junction> const
             clusters.emplace_back(std::move(partition));
             continue;
         }
+        if (partition_size > 200)
```

Comment:
Why is there a hard coded limit at 200?

Reply:
The limit sets the maximum partition size that is still feasible to cluster in a reasonable time. 200 has been a good trade-off between reducing runtime and keeping as many junctions as possible.

It would be nice to replace this hard-coded value with a nice CLI parameter but as it is not a very important value I decided against it. We already have more than 15 parameters and more will be added in the future.

Is there an alternative to adding CLI parameters for every hard-coded value?

Reply:
I see.
Maybe put a `const size_t max_partition_size = 200` with a comment infront of the `if` in line 139 and replace all the 200 with `max_partition_size`, so if someone wanted to change it in code, they could do it by only touching one line.
Also, do you think the user should be informed that the algorithm is behaving differently?
Maybe a `debug_stream << "The number of partition have exceeded " << max_partition_size << ". Applying subsampling to reduce number of partitions\n";`?

Reply:
Great suggestions 👍 I incorporated both of them together with a new test case to keep up the good coverage.